tool/tsh/common/git_config.go

```diff
@@ -0,0 +1,178 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package common
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/gravitational/trace"
+)
+
+// gitConfigCommand implements `tsh git config`.
+//
+// This command internally executes `git` commands like `git config xxx`. You
+// can generally assume the user has `git` binary installed (otherwise there is
+// no point using the `git` proxy feature). An alternative is to use the
```

Comment:
The dependency on `git` binaries isn't completely obvious. I think it is worth giving a clear error message designed to catch this issue.

The other day I was installing VSCode on a fresh machine and it clearly shown a custom error message designed to catch exactly this: lack of `git` binaries. I really appreciated that.